### PR TITLE
Collapse HTTP/2 channel handlers.

### PR DESCRIPTION
**Pull request by Lukasa**
Motivation:

We attempted to use the NIO channel pipeline to spread out the HTTP/2
implementation across a number of channel handlers. Unfortunately, due
to the concerns of reentrancy it's not really safely possible to do this:
it's too easy for the pipeline ordering to become so confused that the
implementation ends up incorrect.

For this reason, we are going to squash together the three handlers that
provide a complete HTTP/2 protocol implementation into just one.

Modifications:

- Move the ConcurrentStreamsHandler and FlowControlHandler into the
    HTTP2Handler.
- Update setup and test code accordingly.

Result:

More correct implementation

### CODE REVIEW

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

File: Tests/NIOHTTP2Tests/OutboundFlowControlBufferTests.swift [hunk 1]
```diff
@@ -0,0 +1,364 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
```
Comment (weissi) [inline Tests/NIOHTTP2Tests/OutboundFlowControlBufferTests.swift]: copyright correct here?

**Comment by Lukasa** [comment]
Blocked by #54.

**Comment by Lukasa** [comment]
Ok, feedback addressed.

**Comment by Lukasa** [comment]
Requires #56.

**Pull request merged**